### ISSUE TITLE
AssertionError in generate_mutations_from

### ISSUE THREAD

**Issue by wsanchez** [HypothesisWorks/hypothesis#3990]
In [CI builds](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202), `dependabot` is trying to upgrade to `hypothesis-6.100.8` and getting a new `AssertionError`:

```console
Traceback (most recent call last):
[918](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:919)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/ims/store/export/test/test_json.py", line 162, in test_fromIO
[919](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:920)
    def test_fromIO(self, imsDataIn: IMSData) -> None:
[920](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:921)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/hypothesis/core.py", line 1638, in wrapped_test
[921](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:922)
    raise the_error_hypothesis_found
[922](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:923)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/hypothesis/core.py", line 1601, in wrapped_test
[923](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:924)
    state.run_engine()
[924](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:925)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/hypothesis/core.py", line 1133, in run_engine
[925](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:926)
    runner.run()
[926](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:927)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/hypothesis/internal/conjecture/engine.py", line 693, in run
[927](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:928)
    self._run()
[928](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:929)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/hypothesis/internal/conjecture/engine.py", line 1142, in _run
[929](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:930)
    self.generate_new_examples()
[930](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:931)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/hypothesis/internal/conjecture/engine.py", line 987, in generate_new_examples
[931](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:932)
    self.generate_mutations_from(data)
[932](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:933)
  File "/home/runner/work/ranger-ims-server/ranger-ims-server/.tox/coverage-py312/lib/python3.12/site-packages/hypothesis/internal/conjecture/engine.py", line 1075, in generate_mutations_from
[933](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:934)
    assert isinstance(new_data, ConjectureResult)
[934](https://github.com/burningmantech/ranger-ims-server/actions/runs/9057012188/job/24880363816?pr=1202#step:11:935)
builtins.AssertionError:
```

**Comment by wsanchez** [comment]
I'm using `@composite` strategies, which may be spitting out something invalid, but it's hard to see how to track that down.

The [strategy in question](https://github.com/burningmantech/ranger-ims-server/blob/master/src/ims/model/strategies.py#L314) here is creating a deeply nested data structure; basically a test database worth of stuff.

**Comment by tybug** [comment]
This should be fixed in #3988 / 6.102.1!

**Comment by Zac-HD** [comment]
Closing because we think it's fixed; if you can repro on the new version we'll reopen 🙂
